core/coredag/dagtransl.go

```diff
@@ -0,0 +1,81 @@
+package coredag
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+
+	node "gx/ipfs/QmUBtPvHKFAX43XMsyxsYpMi3U5VwZ4jYFTo4kFhvAR33G/go-ipld-format"
+	ipldcbor "gx/ipfs/QmYAfx21gPrN2hxNUmsuP6GqLWpxPn351st6rRH3fNuiMU/go-ipld-cbor"
+)
+
+type DagParser func(r io.Reader) ([]node.Node, error)
+
+type FormatParsers map[string]DagParser
+type InputEncParsers map[string]FormatParsers
+
+var DefaultInputEncParsers = InputEncParsers{
+	"json": DefaultJsonParsers,
+	"raw":  DefaultRawParsers,
+}
```

Comment:
If we dont use a global, registration becomes a much more complicated thing to do. Any ideas on a good way to do that?